### ISSUE TITLE
Web UI handling of nested role-based access requests

### ISSUE THREAD

**Issue by zmb3** [gravitational/teleport#15730]
As part of https://github.com/gravitational/teleport/issues/14432 we discovered a tiny corner case with role access requests.

For the following scenario:

- User submits two separate role access requests, `req-1` and `req-2`. 
  - `req-1` provides access to kube cluster `apple`
  - `req-2` provides access to kube cluster `banana` 
- Both requests are approved.
- User assumes the roles granted by `req-1`.
- From this context, user assumes the roles granted by `req-2`.

Because role access requests are additive, the web UI knows that the user is in a context where both `req-1` and `req-2` are active. `tsh` login only supports a single `--request-id` flag.

The most correct behavior would be for the web UI to suggest `tsh login --request-id req-1` when the user clicks connect on kube cluster `apple`, and `tsh login --request-id req-2` when the user clicks connect on kube cluster `banana`. The web UI doesn't know enough to know which resources are granted by which requests, so the actual behavior is that it selects the most recently assumed request, `req-2` in this case (which is correct for `banana` but incorrect for `apple`).

Note that this is not an issue for *resource based access requests*, as those are not additive and the most recently assumed request always overwrites the context from previously assumed requests.

**Comment by xinding33** [comment]
@zmb3 I agree with your suggestion, we should limit the web UI such that users can only assume a single approved role access request at a time.

There are other user experience challenges with our JIT access system (e.g. role access requests are additive while resource access requests are not) but we should address them at a later time.
